### ISSUE TITLE
Unable to find vcvarsall.bat

### ISSUE THREAD

**Issue by kev5** [cocodataset/cocoapi#150]
Windows 10, 64-bit

Getting the following error during the `make` command inside the 'cocoapi/PythoAPI/' directory.

```
python setup.py build_ext --inplace
Compiling pycocotools/_mask.pyx because it changed.
[1/1] Cythonizing pycocotools/_mask.pyx
running build_ext
building 'pycocotools._mask' extension
error: Unable to find vcvarsall.bat
make: *** [all] Error 1
```

**Comment by siddharthkale97** [comment]
Hey, I searched for this error as I am getting the same. I found out many others also got the same problem. And they Installed Visual Studio 2017 and the problem got fixed for them. I was unable to install VS 17 for some reasons, but you should definitely try this out.

**Comment by Samjith888** [comment]
I got the same error as you mentioned above. Every where mentioned to  install Visual studiofor correcting the error, but why i should install Visual studion unless I'm not using it for any project purpose

**Comment by siddharthkale97** [comment]
> I got the same error as you mentioned above. Every where mentioned to install Visual studiofor correcting the error, but why i should install Visual studion unless I'm not using it for any project purpose

I searched on the net and found these exact words on Microsoft site
When you see “unable to find vcvarsall.bat”, it means you’re installing a package that has an extension module, but only the source code. “vcvarsall.bat” is part of the compiler in Visual Studio that is necessary to compile the module.
So you got to install Visual studio else problem like may keep occurring.
link where I found out why ----
https://devblogs.microsoft.com/python/unable-to-find-vcvarsall-bat/
